parser.py

```diff
@@ -404,26 +404,36 @@ def parse_tweets(input_filenames, username, users, html_template, archive_media_
                                         media_sources, users))
     tweets.sort(key=lambda tup: tup[0]) # oldest first
 
-    # Group tweets by month (for markdown)
-    grouped_tweets_markdown = defaultdict(list)
-    for timestamp, md, _ in tweets:
-        # Use a markdown filename that can be imported into Jekyll: YYYY-MM-DD-your-title-here.md
+    # Group tweets by month
+    grouped_tweets = defaultdict(list)
+    for timestamp, md, html in tweets:
+        # Use a (markdown) filename that can be imported into Jekyll: YYYY-MM-DD-your-title-here.md
         dt = datetime.datetime.fromtimestamp(timestamp)
-        markdown_filename = f'{dt.year}-{dt.month:02}-01-Tweet-Archive-{dt.year}-{dt.month:02}.md' # change to group by day or year or timestamp
-        grouped_tweets_markdown[markdown_filename].append(md)
+        filename = f'{dt.year}-{dt.month:02}-01-Tweet-Archive-{dt.year}-{dt.month:02}' # change to group by day or year or timestamp
+        grouped_tweets[filename].append((md, html))
 
-    # Write into *.md files
-    for filename, md in grouped_tweets_markdown.items():
-        md_string =  '\n\n----\n\n'.join(md)
-        with open(filename, 'w', encoding='utf-8') as f:
+    user_input = input(f'Group the html output by month? [y/n]')
+    split_html = user_input.lower() in ('y', 'yes')
+
+    for filename, content in grouped_tweets.items():
```

Comment:
`for filename, (md, html) in grouped_tweets.items():`

Reply:
`grouped_tweets` contains a list of tuples so this wouldn't work.

I can change it into a tuple of lists
```python
grouped_tweets = defaultdict(lambda : (list, list))
for timestamp, md, html in tweets[:2]:
    # [...]
    grouped_tweets[filename][0].append(md)
    grouped_tweets[filename][1].append(html)
```
Or change the loop for something like
`for filename, (md, html) in zip(grouped_tweets.keys(), map(lambda el : zip(*el), grouped_tweets.values())):`

Reply:
Oh sorry, you're quite right.

I think I prefer your original approach but maybe with this style in the joins:

`md, _ for md, html in content`